### PR TITLE
[Merged by Bors] - feat(Topology/Connected/PathConnected): some lemmas for `pathComponentIn`

### PR DESCRIPTION
**Pull request by peabrainiac**
Adds some basic lemmas about `pathComponentIn`, analogous to existing lemmas about `pathComponent` and `connectedComponentIn`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — COMMENTED** [review]
Thanks for working on this corner of the library! These lemmas look good to me (but I don't know that part of the library very well).
I have a few style comments to make, see below. Once you fixed them, I will happily `maintainer merge` this PR.

**Review by grunweg — COMMENTED** [review]
Thanks!
maintainer merge

**Comment by grunweg** [comment]
Thanks for the quick adjustments! To make future reviewing easier, can you (the next time)
- remove the awaiting-author label when you have addressed all comments (otherwise, the PR might be visible as ready for review)
- resolve any review comments you consider addressed? Thanks!

**Comment by grunweg** [comment]
Thanks for the quick adjustments! To make future reviewing easier, can you (the next time)
- remove the awaiting-author label when you have addressed all comments (otherwise, the PR might be visible as ready for review)
- resolve any review comments you consider addressed? Thanks!

**Comment by peabrainiac** [comment]
Oh, sure - I just wanted to get those changes made to the other PR too first, and then got distracted while doing so 😅

But thanks for the reminder, I wasn't sure whether I should resolve the review comments myself or wait for you to take another look.

File: Mathlib/Topology/Connected/PathConnected.lean [hunk 1]
```diff
@@ -832,6 +832,23 @@ theorem Joined.mem_pathComponent (hyz : Joined y z) (hxy : y ∈ pathComponent x
     z ∈ pathComponent x :=
   hxy.trans hyz
 
+theorem mem_pathComponentIn_self (h : x ∈ F) : x ∈ pathComponentIn x F :=
+  JoinedIn.refl h
+
+theorem pathComponentIn_subset : pathComponentIn x F ⊆ F :=
+  fun _ hy ↦ hy.target_mem
+
+theorem pathComponentIn_nonEmpty_iff : (pathComponentIn x F).Nonempty ↔ x ∈ F :=
```
Comment (urkud) [inline Mathlib/Topology/Connected/PathConnected.lean:841]:
```suggestion
theorem pathComponentIn_nonempty_iff : (pathComponentIn x F).Nonempty ↔ x ∈ F :=
```

File: Mathlib/Topology/Connected/PathConnected.lean [hunk 2]
```diff
@@ -832,6 +832,23 @@ theorem Joined.mem_pathComponent (hyz : Joined y z) (hxy : y ∈ pathComponent x
...
+  ⟨fun ⟨_, ⟨γ, hγ⟩⟩ ↦ γ.source ▸ hγ 0, fun hx ↦ ⟨x, mem_pathComponentIn_self hx⟩⟩
+
+theorem pathComponentIn_congr (h : x ∈ pathComponentIn y F) :
+    pathComponentIn x F = pathComponentIn y F := by
+  ext; exact ⟨h.trans, h.symm.trans⟩
+
+theorem pathComponentIn_mono {G : Set X} (h : F ⊆ G) :
```
Comment (urkud) [inline Mathlib/Topology/Connected/PathConnected.lean:848]:
```suggestion
@[gcongr]
theorem pathComponentIn_mono {G : Set X} (h : F ⊆ G) :
```

File: Mathlib/Topology/Connected/PathConnected.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Topology/Connected/PathConnected.lean:841]: Should this be a `@[simp]` lemma?

File: Mathlib/Topology/Connected/PathConnected.lean [hunk 3]
```diff
@@ -832,6 +832,23 @@ theorem Joined.mem_pathComponent (hyz : Joined y z) (hxy : y ∈ pathComponent x
     z ∈ pathComponent x :=
   hxy.trans hyz
 
+theorem mem_pathComponentIn_self (h : x ∈ F) : x ∈ pathComponentIn x F :=
```
Comment (urkud) [inline Mathlib/Topology/Connected/PathConnected.lean:835]: `@[simp]`?

**Comment by peabrainiac** [comment]
As for the suggested `@[simp]` tags, I'm not sure I understand the tactic well enough to accurately judge whether adding those make sense - at least in the case of `mem_pathComponentIn_self`, that would require changing the statement of the lemma to an iff, right? Since simp can only rewrite, not automatically resolve hypotheses like `x ∈ F`?

I could do that and also add a simp tag to `pathComponentIn_nonempty_iff` if you want, but I don't know if they will come up often enough for that to be preferred. The analogous lemmas for `connectedComponentIn` also don't have simp tags yet.

**Comment by grunweg** [comment]
Can you merge master, please (just to make sure the diff looks right after the recent PR) - then I'll ping this on the reviewers channel.

**Comment by grunweg** [comment]
Yes, that's perfect! *Knowing* it wasn't touched is just what I needed. I've asked again on zulip.

**Comment by urkud** [comment]
Let's merge it as is. I'll PR `@[simp]` attrs later. `simp` replaces propositions it can prove with `True`.

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**